db/blob/blob_file_completion_callback.h

```diff
@@ -0,0 +1,56 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+#pragma once
+
+#include "db/error_handler.h"
+#include "file/sst_file_manager_impl.h"
+#include "rocksdb/status.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+class BlobFileCompletionCallback {
+ public:
+  BlobFileCompletionCallback(SstFileManager* sst_file_manager,
+                             InstrumentedMutex* mutex,
+                             ErrorHandler* error_handler)
+      : sst_file_manager_(sst_file_manager),
+        mutex_(mutex),
+        error_handler_(error_handler) {}
+
+  Status OnBlobFileCompleted(const std::string& file_name) {
+    Status s;
+
+#ifndef ROCKSDB_LITE
+    auto sfm = static_cast<SstFileManagerImpl*>(sst_file_manager_);
+    if (sfm) {
+      // Report new blob files to SstFileManagerImpl
+      // TODO (PR7798).  We should only add the file to the FileManager if it
+      // exists. Otherwise, some tests may fail.  Ignore the error in the
+      // interim.
+      sfm->OnAddFile(file_name).PermitUncheckedError();
```

Comment:
Just curious, under what conditions would `OnAddFile` fail here? (When would we call `OnAddFile` for a non-existent file?)

Reply:
I don't think we will call for non existent file but this TODO has been added at other places when call ```OnAddFile``` so to make it consistent I added here. But I can remove this TODO if it is not required here.

Reply:
I _think_ that the reason this is done elsewhere is because `BuildTable` might end up "building" an empty table, in which case it returns `Status::OK()` but there's no file to add to the `sfm`. (Checking this condition would most likely be the right fix for those TODOs BTW.)

This scenario is not possible for `BlobFileCompletionCallback` (since it's only called if a file was in the process of being written), so I think it would be better to go back to `s = sfm->OnAddFile(file_name);` here.

Reply:
Ah Makes Sense. Thanks!